features/steps/argonaut.py

```diff
@@ -344,3 +359,69 @@ def step_impl(context, field_name, value):
                 resource=res,
                 values=found,
                 value=value)
+
+
+def vital_unit_validation(field_name, resource, system_url):
+
+    path = field_name.split('.')
+    path.pop(0)
+
+    systems_to_validate = utils.traverse(resource, path + ["system"])
+    codes_to_validate = utils.traverse(resource, path + ["code"])
+    values_to_validate = utils.traverse(resource, ["component"])
+
+    if not isinstance(systems_to_validate, list):
+        systems_to_validate = [systems_to_validate]
+
+    if not isinstance(codes_to_validate, list):
+        codes_to_validate = [codes_to_validate]
+
+    if not values_to_validate:
+        values_to_validate = [resource]
+
+    if any(system != system_url for system in systems_to_validate):
+        return {"resource": resource, "status": "Wrong System"}
+
+    if any(not in_value_set({"code": code}, system_url) for code in codes_to_validate):
+        return {"resource": resource, "status": "Invalid Code"}
+
+    for value in values_to_validate:
+
+        current_code = value["code"]["coding"][0]["code"]
```

Comment:
This is going to break in cases where there are multiple codings (which is allowed) and the "expected" one doesn't come first. Can we update to

```
codes_with_unit_requirements = c["code"] for c in value["code"]["coding"] if c["code"] in vitals_code_lookup
```

Reply:
We can add another issue to have this "collect" errors instead of returning after the first. A problem in many places of the test suite.